core/CMakeLists.txt

```diff
@@ -25,7 +25,7 @@ set(PRIVATE_LIBS
     stb
 )
 
-add_library(knoting ${KNOTING_SOURCES} ${KNOTING_SHADERS})
+add_library(knoting ${KNOTING_SOURCES} ${KNOTING_SHADERS} "inc/knoting/physics.h")
```

Comment:
In CLion you need to untick the option to add the new file to the CMakeLists.txt. I understand you are using Visual Studio, and do not know if there is any similar option.